huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go

```diff
@@ -0,0 +1,366 @@
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hbase_public_endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"open_tsdb_link": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"opentsdb_public_endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"storage_size": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"used_storage_size": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"zookeeper_link": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func buildCloudTableClusterParams(d *schema.ResourceData) clusters.CreateOpts {
+	result := clusters.CreateOpts{
+		Name:           d.Get("name").(string),
+		VpcId:          d.Get("vpc_id").(string),
+		StorageType:    d.Get("storage_type").(string),
+		IAMAuthEnabled: d.Get("iam_auth_enabled").(bool),
+		Datastore: clusters.Datastore{
+			Type:    "hbase",
+			Version: "1.0.6",
+		},
+		Instance: clusters.Instance{
+			AvailabilityZone: d.Get("availability_zone").(string),
+			CUNum:            d.Get("rs_num").(int),
+			Networks: []clusters.Network{
+				{
+					SubnetId:        d.Get("network_id").(string),
+					SecurityGroupId: d.Get("security_group_id").(string),
+				},
+			},
+		},
+	}
+	if num, ok := d.GetOk("opentsdb_num"); ok {
+		result.OpenTSDBEnabled = true
+		result.Instance.TSDNum = num.(int)
+	}
+
+	return result
+}
+
+func resourceCloudTableClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.CloudtableV2Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud CloudTable V2 client: %s", err)
+	}
+
+	opt := buildCloudTableClusterParams(d)
+	resp, err := clusters.Create(client, opt)
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud CloudTable cluster: %s", err)
+	}
+	d.SetId(resp.ClusterId)
+	stateRef := stateRefresh{
+		Pending:      []string{"Creating"},
+		Target:       []string{"Running"},
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        20 * time.Second,
+		PollInterval: 15 * time.Second,
+		// Due to the design of the API, the cluster cannot be queried for a few time after the create request is sent.
+		NotFoundChecks: 1,
+	}
+	if err := waitForCloudTableClusterStateRefresh(ctx, client, d.Id(), stateRef); err != nil {
+		return err
+	}
+
+	return resourceCloudTableClusterRead(ctx, d, meta)
+}
+
+func setCloudTableClusterStatus(d *schema.ResourceData, strCode string) error {
+	if val, ok := stateCodes[strCode]; ok {
+		return d.Set("status", val)
+	}
+	return fmtp.Errorf("the status is abnormal, the code is %+v")
+}
+
+func setCloudTableClusterTSDNum(d *schema.ResourceData, strNum string) error {
+	tsdNum, err := strconv.Atoi(strNum)
+	if err != nil {
```

Comment:
do not raise error when converting failed, just set the field when err is nil

Reply:
done